asapdiscovery-docking/asapdiscovery/docking/scripts/run_docking_oe.py

```diff
@@ -404,9 +479,14 @@ def main():
         "SMILES",
     ]
     nprocs = min(mp.cpu_count(), len(mp_args), args.num_cores)
-    print(f"Running {len(mp_args)} docking runs over {nprocs} cores.")
-    with mp.Pool(processes=nprocs) as pool:
-        results_df = pool.starmap(mp_func, mp_args)
+    if nprocs == 0:
+        print(f"Running {len(mp_args)} docking runs sequentially.")
+        results_df = dock_sequentially(mp_args)
+
+    else:
+        print(f"Running {len(mp_args)} docking runs over {nprocs} cores.")
+        with mp.Pool(processes=nprocs) as pool:
+            results_df = pool.starmap(mp_func, mp_args)
```

Comment:
I may be missing something, but I think we can just use `mp_func` in the non-multiprocessed version as well, eg
```suggestion
    if nprocs == 0:
        print(f"Running {len(mp_args)} docking runs sequentially.")
        results_df = [mp_func(*a) for a in mp_args]
    else:
        print(f"Running {len(mp_args)} docking runs over {nprocs} cores.")
        with mp.Pool(processes=nprocs) as pool:
            results_df = pool.starmap(mp_func, mp_args)
```

and then get rid of the `dock_sequentially` function

Reply:
I agree if it is a one for one match, but lets rename it to `dock_something` to make it clear what the function is doing. `mp_func ` is not clear.